extensions/amp-story/0.1/amp-story-auto-ads.js

```diff
@@ -89,11 +99,56 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    this.readConfig_();
     this.schedulePage_();
     return Promise.resolve();
   }
 
 
+  /**
+   * load in config from child <script> element
+   * @private
+   */
+  readConfig_() {
+    const children = this.element.children;
+    dev().assert(children.length >= 1, `<${TAG}>: this tag's first child` +
+        'should be a <script> element');
+
+    const child = children[0];
+    user().assert(
+        isJsonScriptTag(child),
+        `The <${TAG}> config should ` +
+        'be inside a <script> tag with type="application/json"');
+
+    try {
+      this.config_ = parseJson(child.textContent);
+    } catch (e) {
+      user().error(TAG, 'Invalid JSON config', e);
+    }
+
+    this.validateConfig_();
+  }
+
+
+  /**
+   * make sure given JSON config is shaped correctly
+   * @private
+   */
+  validateConfig_() {
+    const adAttributes = this.config_['ad-attributes'];
+    user().assert(adAttributes, `<${TAG}>: Error reading config.` +
+      'Top level JSON should have an "ad-attributes" key');
+
+    const type = adAttributes.type;
+    user().assert(type, `<${TAG}>: Error reading config.` +
+      'Missing ["ad-attribues"]["type"] key');
+
+    const dataSrc = adAttributes['data-src'];
```

Comment:
this is only a must have for custom ad. you can remove this check

Reply:
removed